### DISCUSSION TITLE
I'm facing a problem in solving for the duals

### DISCUSSION THREAD

**Discussion by godunable** [yalmip/YALMIP#1361] [General discussion]
I am solving a simple optimization problem and finding its dual. However, when I use the dual function to solve it, the result is NaN. I guess this is due to the fact that my objective function and constraints contain non-linear functions (e.g. 1/x)  or I'm using a nonlinear solver. I'm not sure if my guess is correct, and if it is, how do I find the dual of the problem?

f = sdpvar(10,1);
F = 1;
T = 4;
C = 0.12*ones(10,1); 

T_a = C./f; 

Constraints=[];
Constraints=[Constraints,f<=F];
Constraints=[Constraints,f>=0];
Constraints=[Constraints,T_a<=T];

T_all=sum(T_a);

options = sdpsettings('solver','mosek');
optimize(Constraints,T_all,options)
a1=dual(Constraints(1));
a2=dual(Constraints(2));
a3=dual(Constraints(3));

![20231214170000](https://github.com/yalmip/YALMIP/assets/42164173/a02c321d-d088-4e65-8f9a-b68c34080310)

**Comment by johanlofberg** [comment]
This nonlinear model requires reformulations by YALMIP to write it as something that Mosek can handle, and when doing so duals are not recovered 

You will have to manually write the objective using SOCPs via the cone operator to obtain a model in native Mosek format to enable dual recovery

**Reply by godunable** [reply]
Thanks to your reply, I think it's time for me to learn about SOCP. Actually, that question above is only part of my problem, my problem also includes some exponential cones, for example, my objective function is:
T_all=sum(1./log2(1+f)),
 in that case can it be solved by SOCP too?

**Reply by johanlofberg** [reply]
Objective 1/log(1+f) means you introduce variable t with constraint 1/log(1+f) <= t i.e. (1/t) <= log(1+f) which is convex and exponential cone representable (in YALMIP you just write cpower(t,-1) <= log(1+f) and the expcone model will be derived but if you need the duals you will manually have to derive the SOCP model of 1/t using one more epigraph variables and then the cone representation of 1/t<=s)
